server/services/lduService.js

```diff
@@ -0,0 +1,56 @@
+const logger = require('../../log')
+/**
+ * @typedef {import("../../types/licences").LduService} LduService
+ * @typedef {import("../../types/licences").ActiveLduClient} lduActiveClient
+ * @typedef {import("../../types/delius").DeliusClient} DeliusClient
+ */
+
+/**
+ * @param {DeliusClient} deliusClient
+ * @return {LduService}
+ */
+
+module.exports = function createLduService(deliusClient, lduActiveClient) {
+  return {
+    async getAllProbationAreas() {
+      return deliusClient.getAllProbationAreas()
+    },
+
+    async getLdusForProbationArea(probationAreaCode) {
+      return deliusClient.getAllLdusForProbationArea(probationAreaCode)
+    },
+
+    async getActiveLdusForProbationArea(probationAreaCode) {
+      return lduActiveClient.allActiveLdusInArea(probationAreaCode)
+    },
+
+    async updateActiveLdus(probationAreaCode, activeLdus) {
+      // to update the active_local_delivery_units with only active LDUs
+      const successLogMsg = 'active_local_delivery_units table updated with user selected LDUs'
+      const failureLogMsg = 'Could not update active_local_delivery_units with user selected LDUs'
+      let activeLdusArray = activeLdus
+      let activeLduCodes
+
+      if (!Array.isArray(activeLdus)) {
```

Comment:
you should be able to ensure that this is always provided from the frontend as an array - when doing the frontend just need to assign indexes to the parameter names and it will filter out the empty values.

Reply:
but this is a number of checkboxes not a dropdown selection. How will I assign an index to them?

Pus is it better to have logic in backend service only?

Reply:
discuss